tools/travis-scripts/run-script.sh

```diff
@@ -44,7 +44,8 @@ function build_linux_clang_tidy()
     mkdir -p clang-tidy-build
     cd clang-tidy-build
     cmake ../.. -DCMAKE_EXPORT_COMPILE_COMMANDS=on
-    python $(dirname $(dirname $(readlink -f `which clang-tidy`)))/share/clang/run-clang-tidy.py
+    clang_tidy_script=$(dirname $(dirname $(readlink -f `which clang-tidy`)))/share/clang/run-clang-tidy.py 
+    python $clang_tidy_script '^((?!/cocos2d-x/external/).)*$'
```

Comment:
could you please add some comment here?

Reply:
`'^((?!/cocos2d-x/external/).)*$'` is a python regular expression. It's used in `run-clang-tidy.py`,

```python
file_name_re = re.compile('|'.join(args.files))
```
with `args.files = '^((?!/cocos2d-x/external/).)*$'`. 

The regular expression is used to check all files included in the compilation database. If a file path contains `/cocos2d-x/external/`, that file will be ignored for `clang-tidy`.

Reference: [Python regular expression documentation](https://docs.python.org/2.7/library/re.html)


Here is a small test case for the python re used,

```python
import re

files = [ 
"/home/john/cocos2d-x/external/openssl/include/linux/openssl/bio.h",
"/home/john/cocos2d-x/external/tiff/include/linux/tiff.h",
"/home/john/cocos2d-x/external/json/stringbuffer.h",
"/home/john/cocos2d-x/cocos/base/ccUtils.h",
"/home/john/cocos2d-x/cocos/physics/CCPhysicsBody.cpp",
"/home/john/cocos2d-x/tests/cpp-tests/Classes/ActionsEaseTest/ActionsEaseTest.cpp",
"/home/john/cocos2d-x/templates/cpp-template-default/Classes/AppDelegate.cpp"
]

pattern = '^((?!/cocos2d-x/external/).)*$'

for file in files:
    m = re.search(pattern, file)
    if m:
        print m.group(0)
```

The output shows all file includes `/cocos2d-x/external/` are filtered out.
```bash
/home/john/cocos2d-x/cocos/base/ccUtils.h
/home/john/cocos2d-x/cocos/physics/CCPhysicsBody.cpp
/home/john/cocos2d-x/tests/cpp-tests/Classes/ActionsEaseTest/ActionsEaseTest.cpp
/home/john/cocos2d-x/templates/cpp-template-default/Classes/AppDelegate.cpp
```

Reply:
Thanks for the explanation, what i mean is to add comments in the script file, because regexp is not readable.

Reply:
added.

Reply:
thanks